### ISSUE TITLE
MetaMaskSdk's provider don't Resolve promise request({method: 'eth_requestAccounts'})

### ISSUE THREAD

**Issue by sensactive** [MetaMask/metamask-sdk#344]
in my case i am using a browser application. My scenario:

const ethereum = new MetaMaskSDK().getProvider()
try {
await ethereum.request({ method:
'eth_requestAccounts' })
console log('connected')
} catch (error) {
console.log('error', error)
}
Actual result:
None of the output is displayed in the console log. The promise remains in pending status all the time.

Expected Result:
Any of the console logs is displayed, depending on the user's action in the MetaMask mobile application

Detection Environment: Google Chrome 112.0.5615.49
Device: Samsung Galaxy Note 20 Ultra, Android 13, One UI 5.1, App version 6.3.0

P.S
I also want to add that this problem concerns only the mobile application, when you try to perform this action in the browser extension - everything works as expected, i.e. Promise changes its state to resolve or rejected. It is also noticed that after the user has confirmed or not confirmed the connection in the mobile application, no incoming messages are received in the open web socket connection, this is probably the problem

**Comment by abretonc7s** [comment]
Hi @sensactive ,
which version of the sdk are you using?

You should initialize with:
```ts
const sdk = new MetaMaskSDK({dappMetadata: { url: '...', title: '...'} }) 
await sdk.init()
const provider = sdk.getProvider()
```

Please take a look at the example folder as well

**Comment by sensactive** [comment]
Hi, @abretonc7s ! My version SDK is 0.1.0.

I don't understand why I should use the example you suggested. I need to get two way communication from mobile browser to my mobile app. I’m just using an example from the documentation, I wrote it above, just in case I’ll attach a screenshot. I don't have data for dappMetadata(such as url and title), I only have mobile app which is connected to testnet. Just in case, I repeat: my version of the code works perfectly with a browser extension (i.e. on a desktop computer and on a desktop browser)

**Comment by sensactive** [comment]
![image](https://github.com/MetaMask/metamask-sdk/assets/36218253/8fa3e263-dfd2-4ee9-84a2-e0061a8fd3ee)

**Comment by christopherferreira9** [comment]
Hi @sensactive ! Can you please try the latest version of the [pure-js example](https://github.com/MetaMask/metamask-sdk/tree/main/packages/examples/pure-javascript) that we have in our repo? Also, the cdn link we include in the documentation has just been updated to the newest version of the MetaMask SDK.

**Comment by christopherferreira9** [comment]
Hi @sensactive , have you managed to check our example to see if it works?

**Comment by sensactive** [comment]
Hi @christopherferreira9! I was a little misled about the use case. I'm sorry. I use metamask-sdk in my project on the Vue 3 framework. In the repository examples, I found an example just for Vue 3, but it works in a desktop browser; I have not yet been able to run this solution in a mobile browser. I need a little more time to check in the mobile browser. I will definitely post the results here, as this is very important to me.

PS I updated my npm package to version 0.7.0
